lib/freebl/algcmac.c

```diff
@@ -0,0 +1,327 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifdef FREEBL_NO_DEPEND
+#include "stubs.h"
+#endif
+
+#include "secport.h"
+#include "blapit.h"
+#include "blapi.h"
+#include "algcmac.h"
+#include "secerr.h"
+#include "nspr.h"
+
+#include "stdio.h"
+
+// Enum for identifying the underlying block cipher we're using internally.
+// To add a new cipher, add an entry here, define new CMAC_{name}Init and
+// CMAC_{name}Create methods, add a new Context pointer to the cipher union,
+// and update the cmac_Encrypt and CMAC_Destroy methods to handle the new
+// cipher.
```

Comment:
I prefer to try to keep c++ style comments out of C code. There was a time where this was a hard requirement, but I think all our platforms accepts c++ style comments. Still it's best to not mix them. (not a hard requirement as upstream regularly adds these style comments).

Reply:
Ah sorry, will do.

Reply:
Ah never mind, I [found it](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/NSS_Developer_Tutorial).